diff --git forkSrcPrefix/cmake/onnxruntime.cmake forkDstPrefix/cmake/onnxruntime.cmake
index c900f4d4b09a5330683912a6f7d1cb33fc8ffdc9..d76cb31620633666ffa428351c5c845a43ef8b22 100644
--- forkSrcPrefix/cmake/onnxruntime.cmake
+++ forkDstPrefix/cmake/onnxruntime.cmake
@@ -228,6 +228,20 @@ target_link_libraries(onnxruntime PRIVATE
     ${onnxruntime_EXTERNAL_LIBRARIES}
 )
 
+# Arch-dependent Link options:
+message("CMAKE_CXX_COMPILER_ID = ${CMAKE_CXX_COMPILER_ID}")
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    target_link_options(onnxruntime PUBLIC "-Wl,-map,onnxruntime.map")
+  else()
+    target_link_options(onnxruntime PUBLIC LINKER:-Map,onnxruntime.map)
+  endif()
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  target_link_options(onnxruntime PUBLIC LINKER:-Map,onnxruntime.map)
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  target_link_options(onnxruntime PRIVATE "/MAP:onnxruntime.map")
+endif()
+
 set_property(TARGET onnxruntime APPEND_STRING PROPERTY LINK_FLAGS ${ONNXRUNTIME_SO_LINK_FLAG} ${onnxruntime_DELAYLOAD_FLAGS})
 set_target_properties(onnxruntime PROPERTIES
   PUBLIC_HEADER "${ONNXRUNTIME_PUBLIC_HEADERS}"
